### ISSUE TITLE
[oracle] 'Found unparsable section' with restrict_references pragma

### ISSUE THREAD

**Issue by osarier** [sqlfluff/sqlfluff#7957]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I wanted to run the linter on a file with the following example. This leads to the described error.

### Expected Behaviour

I would expect the example to be correctly parsed. The keyword default should be supported as well. 
See https://docs.oracle.com/en/database/oracle/oracle-database/26/lnpls/RESTRICT_REFERENCES-pragma.html

### Observed Behaviour

> Line 1, Position 1: Found unparsable section: 'create or replace package ex_pkg is/n   p...'

### How to reproduce

```sql
create or replace package ex_pkg is
    pragma restrict_references(default);
end ex_pgk;
``` 

This example seems to be working fine:

```sql
create or replace package ex_pkg is
    pragma restrict_references(ex_func);
end ex_pgk;
``` 

### Dialect

oracle

### Version

4.2.1
python 3.11.3

### Configuration

[sqlfluff]
dialect = oracle

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
